system-contracts/test/L1Messenger.spec.ts

```diff
@@ -0,0 +1,457 @@
+      index: 2,
+      initValue: BigNumber.from(0),
+      finalValue: BigNumber.from(1),
+    },
+  ];
+  const encodedStateDiffs = encodeStateDiffs(stateDiffs);
+  const compressedStateDiffs = compressStateDiffs(4, stateDiffs);
+  const enumerationIndexSizeBytes = ethers.utils.hexZeroPad(ethers.utils.hexlify(4), 1);
+  await setResult(
+    "Compressor",
+    "verifyCompressedStateDiffs",
+    [stateDiffs.length, 4, encodedStateDiffs, compressedStateDiffs],
+    {
+      failure: false,
+      returnData: ethers.utils.defaultAbiCoder.encode(["bytes32"], [ethers.utils.keccak256(encodedStateDiffs)]),
+    }
+  );
+  const numberOfStateDiffsBytes = ethers.utils.hexZeroPad(ethers.utils.hexlify(stateDiffs.length), 4);
+  const compressedStateDiffsSizeBytes = ethers.utils.hexZeroPad(
+    ethers.utils.hexlify(ethers.utils.arrayify(compressedStateDiffs).length),
+    3
+  );
+  return {
+    encodedStateDiffs,
+    compressedStateDiffs,
+    enumerationIndexSizeBytes,
+    numberOfStateDiffsBytes,
+    compressedStateDiffsSizeBytes,
+  };
+}
+
+// The LogData interface represents the structure of the data that will be logged
+interface LogData {
+  isService: boolean;
+  key: Buffer;
+  value: Buffer;
+  message: Buffer;
+  currentMessageLengthBytes: string;
+  logs: string[];
+}
+
+function setupLogData(l1MessengerAccount: ethers.Signer, l1Messenger: L1Messenger): LogData {
+  const key = Buffer.alloc(32, 1);
+  const value = Buffer.alloc(32, 2);
+  const message = Buffer.alloc(32, 3);
+  const currentMessageLengthBytes = ethers.utils.hexZeroPad(ethers.utils.hexlify(32), 4);
+  const logs = [
+    ethers.utils.concat([
+      ethers.utils.hexlify([0]),
+      ethers.utils.hexlify(1),
+      ethers.utils.hexZeroPad(ethers.utils.hexlify(1), 2),
+      ethers.utils.hexZeroPad(l1MessengerAccount.address, 20),
+      key,
+      value,
+    ]),
+    ethers.utils.concat([
+      ethers.utils.hexlify([0]),
+      ethers.utils.hexlify(1),
+      ethers.utils.hexZeroPad(ethers.utils.hexlify(1), 2),
+      ethers.utils.hexZeroPad(l1Messenger.address, 20),
+      ethers.utils.hexZeroPad(ethers.utils.hexStripZeros(l1MessengerAccount.address), 32).toLowerCase(),
+      ethers.utils.keccak256(message),
+    ]),
+  ];
+  return {
+    isService: true,
+    key,
+    value,
+    message,
+    currentMessageLengthBytes,
+    logs,
+  };
+}
+
+//Represents the structure of the bytecode/message data that is part of the pubdata.
+interface BytecodeData {
+  content: string;
+  lengthBytes: string;
+}
+
+async function setupBytecodeData(l1MessengerAddress: string): Promise<BytecodeData> {
+  const content = await getCode(l1MessengerAddress);
+  const lengthBytes = ethers.utils.hexZeroPad(ethers.utils.hexlify(ethers.utils.arrayify(content).length), 4);
+  return {
+    content,
+    lengthBytes,
+  };
+}
+
+// Used for emulating the pubdata published by the L1Messenger.
+class L1MessengerPubdataEmulator {
+  numberOfLogs: number;
+  logs: string[];
```

Comment:
Add a comment that these are not logs, but encodings of logs.

Reply:
fxied